src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs

```diff
@@ -43,5 +43,14 @@ public class AspNetInstrumentationOptions
         /// The type of this object depends on the event, which is given by the above parameter.</para>
         /// </remarks>
         public Action<Activity, string, object> Enrich { get; set; }
+
+        /// <summary>
+        /// Gets or sets an action that uses HttpContext information to set the appropriate activity DisplayName.
+        /// </summary>
+        /// <remarks>
+        /// <para>HttContext: the current HttpContext.</para>
+        /// <para><see cref="Activity"/>: the activity for which DisplayName has to be set.</para>
+        /// </remarks>
+        public Action<HttpContext, Activity> SetActivityDisplayName { get; set; }
```

Comment:
this alone is not an improvement right? We already provide Enrich, which can be used by user to change DisplayName.

Reply:
`SetActivityDisplayName` needs `HttpContext` to set the correct name for the activity. We don't provide `HttpContext` to `Enrich` instead we provide `HttpResponse` which is not enough to derive the correct name for the activity.

Reply:
We should be providing both HttpRequest and HttpResponse to Enrich.. Isn't routeinformation part of HttpRequest?

Reply:
IMO, if this is for specifically for setting activity display name, it should have a signature something like...

```csharp
   public Func<HttpContext, string> DisplayNameOverrideFunc { get; set; }
```

Used something like...

```csharp
   if (options.DisplayNameOverrideFunc != null)
      activity.DisplayName = options.DisplayNameOverrideFunc(httpContext);
```

Reply:
Do we want to add a new option just to have custom logic for setting the Activity DisplayName? Or do we just start providing `HttpContext` to `Enrich` option instead of `HttpResponse`. This way the user could implement their logic to set the Activity DisplayName in `Enrich` itself.

Reply:
I definitely would vote for just having a better Enrich call. What are you thinking `Action<Activity, string, HttpContext, object>`? I've never been a big fan of the `object` in the signature. `Action<Activity, string, HttpContext>` would be better I think, user can use the event name to know to look for request vs response?

Reply:
I was thinking of continuing to use the existing signature: `Action<Activity, string, object>`. I would have preferred to use the specific type instead of `object` but this is the pattern we have followed in every other instrumentation such as Http, Grpc, AspNetCore. We could use a specific type and standardize this for other instrumentations too.

Currently, when `Enrich` is called, we pass `HttpRequest` as the `object` in the `OnStartActivity` method and `HttpResponse`  as the `object` in `OnStopActivity` method. If we want to update the signature to `Action<Activity, string, HttpContext>` we would have to update the `OnStartActivity` method to provide `HttpContext` instead of HttpRequest`.  If we follow this approach, the user will always get an `HttpContext` and based on the event name they can either use `HttpContext.Request` or `HttpContext.Response` or both.

Reply:
> If we follow this approach, the user will always get an `HttpContext` and based on the event name they can either use `HttpContext.Request` or `HttpContext.Response` or both.

Ya I vote for that. But let's see what others think.